api_v3_protocol_parameters.go

```diff
@@ -349,6 +350,12 @@ func congestionControlSanityCheck(protocolParams *V3ProtocolParameters) {
 	}
 }
 
+func rewardsSanityCheck(protocolParams *V3ProtocolParameters) {
+	if math.Log2(float64(protocolParams.TokenSupply()))+float64(protocolParams.RewardsParameters().PoolCoefficientExponent) > 63 {
+		panic("Token supply bits count + poolCoefficientExponent must be less than or equal to 63")
+	}
```

Comment:
```suggestion
    tokenSupplyBitsCount := uint8(math.Log2(float64(protocolParams.TokenSupply())))
	if tokenSupplyBitsCount + protocolParams.RewardsParameters().PoolCoefficientExponent > 63 {
		message := fmt.Sprintf("Token supply bits count (%d) + PoolCoefficientExponent (%d) must be less than or equal to 63\n", tokenSupplyBitsCount, protocolParams.RewardsParameters().PoolCoefficientExponent)
		panic(message)
	}
```

Reply:
Maybe @oliviasaa can comment on whether it should be `> 63` or `>= 63`. I'm not sure right now.

Also I guess if we truncate the float result to integer it would have to be `uint8(log2(tokenSupply)) + 1` in case there is a decimal place?

Reply:
If you use uint8 in this way, then you also need to check for overflows incase someone were to set PoolCoefficentExponent = 255 for example, then the sanity check would pass but this would obviously not be a sane value.

Reply:
I am actually also sceptical about one of the other assumptions in the comments/TIP, which is that the target reward is at most 42 bits. Where does this assumption come from and does it also need to be sanity checked?

Reply:
Regarding the >63 or >=63 point, I actually think the check can be >64 if, as you suggested, we use the actual bit count used by the token supply which is given by uint8(log2(tokenSupply))+1. So the pool stake takes up tokenSupplyBits, and that gets shifted left by poolCoefficientExponent, so that can be up to 64 bits without issue. Still happy to wait for @oliviasaa to check though.

Reply:
<64 :)

Reply:
and I agree with Andrew that using floats for the tests is simpler, because of a potential overflowing

Reply:
>that the target reward is at most 42 bits
this was in the initial calculations, which used the iota supply,but I can provide you with a better check

Reply:
> this was in the initial calculations, which used the iota supply, but you can just check if Computed Initial Rewards uses this amount of bits

@oliviasaa are you saying that the reference to 42 bits is outdated and should be updated to something else? Do we need another check that Computed Initial Rewards uses some amount of bits? The tip should be updated and so should the comment in the code to make this understandable either way.